### ISSUE TITLE
Problems when saving multiple annotations tables for the same element

### ISSUE THREAD

**Issue by LucaMarconato** [scverse/napari-spatialdata#260]
I have observed problems when saving multiple annotation tables for the same shapes element using the annotation widget. Not sure if it's a macOS specific bug @melonora. Here is how to reproduce:

- open `napari-spatialdata` and a layer from `SpatialData` object
- create a new shapes layer
- open the annotation widget
- link the layer to the sdata
- create some classes
- create some annotations around the classes
- save the annotation table

So far all works great.

- add a new annotation class, create a new annotation
- [x] save the annotation using the same element name but a new annotation table. This leads to a warning (something like `Shapes_external and annotation_Shapes_external2 already exist. Do you want to overwrite them?`). The message is incorrect. It should have an "or" instead of the "and".
- [x] If we save (but also if we don't save I think), the combobox "Annotation table:" will add some duplicate entries.
- [x] If I added extra shapes before saving the second time, then going through the entries of the "Annotation table:" combobox would lead to warnings like `UserWarning: The provided face_color parameter has 4 entries, while the data contains 5 entries. Setting face_color to white.!` and to the annotations being displayed as white for some of the entries instead of with the class colors.

Finally:
- [ ] if I save the annotations under a new element and under a new table name, the original shapes layer gets renamed to the new name. Therefore, since we are not really saving a new element but updating the original one, I would consider the following:
	- ~~when "Save selected annotation layer" is pressed, the user can only choose the name of the annotation table~~ edit: we opted for letting the user choose both names
	- [ ] and the shapes layer is always overwritten in place.

**Comment by melonora** [comment]
will check

**Comment by LucaMarconato** [comment]
> and the shapes layer is always overwritten in place.

can you confirm that now with #261 the shapes layer is overwritten in place? At least I see that the old layer disappears from the view and is replaced by the new layer (with eventually a new name). I guess therefore that also on disk the old layer (and old name) disappears. If so, we can tick this last checkbox.

**Comment by melonora** [comment]
see description of #261

**Comment by melonora** [comment]
TLDR yes it is. The old shape will still exist though when the user changed both the table name and the shapes name.
